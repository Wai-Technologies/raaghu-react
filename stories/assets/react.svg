<svg width="84" height="80" viewBox="0 0 84 80" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" fill="#61DAFB"/>
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M42 55.4964C64.0914 55.4964 82 48.5584 82 40C82 31.4415 64.0914 24.5035 42 24.5035C19.9086 24.5035 2 31.4415 2 40C2 48.5584 19.9086 55.4964 42 55.4964Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7734 47.7482C39.8191 67.1601 54.6952 79.4276 62 75.1484C69.3048 70.8692 66.2723 51.6637 55.2266 32.2518C44.1809 12.8398 29.3048 0.572317 22 4.85154C14.6952 9.13076 17.7277 28.3363 28.7734 47.7482Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7732 32.2518C17.7275 51.6637 14.695 70.8692 21.9998 75.1485C29.3046 79.4277 44.1807 67.1602 55.2264 47.7482C66.2721 28.3363 69.3046 9.1308 61.9998 4.85158C54.695 0.572351 39.8189 12.8398 28.7732 32.2518Z" stroke="#61DAFB" stroke-width="4"/>
</svg>
