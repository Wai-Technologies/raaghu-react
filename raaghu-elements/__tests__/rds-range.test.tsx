import React from "react";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import RdsRange, { RdsRangeProps } from "../src/rds-range/rds-range";


describe("RdsRange", () => {
    const defaultProps: RdsRangeProps = {
        min: 0,
        max: 100,
        rangeType: "default",
    };

    it("renders a different type of range when the rangeType prop is set", () => {
        const { container } = render(<RdsRange {...defaultProps} rangeType="type1" />);

        const rangeTwo = container.querySelector(".slider_two");
        expect(rangeTwo).toBeInTheDocument();
    });

    it("renders the component", () => {
        const { container } = render(<RdsRange {...defaultProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it("updates the value on change", () => {
        const { getByRole, getByText } = render(<RdsRange {...defaultProps} />);
        const input = getByRole("slider");
        fireEvent.change(input, { target: { value: "50" } });
        expect(input).toHaveValue("50");
        expect(getByText("50")).toBeInTheDocument();
    });

    it("displays the correct range values", () => {
        const { getAllByText } = render(<RdsRange {...defaultProps} />);
        const rangeValues = getAllByText("0");
        expect(rangeValues).toHaveLength(2);
        expect(getAllByText("100")).toHaveLength(1);
    });
});


