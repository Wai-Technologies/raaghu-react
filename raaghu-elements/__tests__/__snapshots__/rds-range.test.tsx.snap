// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RdsRange renders the component 1`] = `
<div
  class="position-relative py-5"
>
  <div
    class="slider-track"
  >
    <span
      class="sliderTooltipRange"
      id="range1"
      style="top: -55px; padding: 5px 10px;"
    >
      0
    </span>
  </div>
  <input
    class="slider_0 "
    max="100"
    min="0"
    type="range"
    value="0"
  />
  <div
    class="showvalue mt-4"
  >
    <span>
      0
    </span>
     
    <span>
      100
    </span>
  </div>
</div>
`;
